node/full_client.go

```diff
@@ -542,8 +544,19 @@ func (c *FullClient) Validators(ctx context.Context, heightPtr *int64, pagePtr,
 }
 
 // Tx returns detailed information about transaction identified by its hash.
-func (c *FullClient) Tx(ctx context.Context, hash []byte, prove bool) (*ctypes.ResultTx, error) {
+func (c *FullClient) Tx(ctx context.Context, hash []byte, prove bool) (txRes *ctypes.ResultTx, err error) {
+	defer func() {
+		if recoveryError := recover(); recoveryError != nil {
+			if recoveryError == ds.ErrNotFound {
+				err = fmt.Errorf("tx (%X) not found", hash)
+			} else {
+				panic(recoveryError)
+			}
+		}
+	}()
```

Comment:
@tzdybal should we catch panic in `func (txi *TxIndex) Get` or should we just return err in [line 55](https://github.com/vuong177/rollkit/blob/18dc8d32555ecfb754d2327dc3ff2711bc7851f1/state/txindex/kv/kv.go#L55) instead of panic here ?